bser.h

```diff
@@ -0,0 +1,104 @@
+#ifndef _LIBWATCHMAN_BSER_H_
+#define _LIBWATCHMAN_BSER_H_
+
+#include <stdint.h>
+
+#include "bser_private.h"
+
+/* ctors  - if 'fill' is NULL a new object will be allocated and returned,
+   otherwise, fill is initailized and returned */
+bser_t* bser_new_integer(uint64_t value, bser_t* fill);
+bser_t* bser_new_real(double value, bser_t* fill);
+bser_t* bser_new_true(bser_t* fill);
+bser_t* bser_new_false(bser_t* fill);
+bser_t* bser_new_null(bser_t* fill);
+bser_t* bser_new_string(const char* chars, size_t len, bser_t* fill);
+bser_t* bser_new_array(bser_t* elems, size_t len,
+    bser_t* fill);
+bser_t* bser_new_object(bser_key_value_pair_t* fields, size_t length, bser_t* fill);
+
+void bser_free(bser_t* bser);
+
+/* type queries */
+static inline int bser_is_integer(bser_t* bser) {
```

Comment:
don't put these in the header -- the compiler should be smart enough to inline them even if they're in the c file.